tests/08-native-runs/14-sha-256/test-sha-256.c

```diff
@@ -0,0 +1,469 @@
+      0xf0 , 0xf1 , 0xf2 , 0xf3 , 0xf4 , 0xf5 , 0xf6 , 0xf7 ,
+      0xf8 , 0xf9 , 0xfa , 0xfb , 0xfc , 0xfd , 0xfe , 0xff
+    },
+    80,
+    {
+      0x06 , 0xa6 , 0xb8 , 0x8c , 0x58 , 0x53 , 0x36 , 0x1a ,
+      0x06 , 0x10 , 0x4c , 0x9c , 0xeb , 0x35 , 0xb4 , 0x5c ,
+      0xef , 0x76 , 0x00 , 0x14 , 0x90 , 0x46 , 0x71 , 0x01 ,
+      0x4a , 0x19 , 0x3f , 0x40 , 0xc1 , 0x5f , 0xc2 , 0x44
+    },
+    {
+      0xb1 , 0x1e , 0x39 , 0x8d , 0xc8 , 0x03 , 0x27 , 0xa1 ,
+      0xc8 , 0xe7 , 0xf7 , 0x8c , 0x59 , 0x6a , 0x49 , 0x34 ,
+      0x4f , 0x01 , 0x2e , 0xda , 0x2d , 0x4e , 0xfa , 0xd8 ,
+      0xa0 , 0x50 , 0xcc , 0x4c , 0x19 , 0xaf , 0xa9 , 0x7c ,
+      0x59 , 0x04 , 0x5a , 0x99 , 0xca , 0xc7 , 0x82 , 0x72 ,
+      0x71 , 0xcb , 0x41 , 0xc6 , 0x5e , 0x59 , 0x0e , 0x09 ,
+      0xda , 0x32 , 0x75 , 0x60 , 0x0c , 0x2f , 0x09 , 0xb8 ,
+      0x36 , 0x77 , 0x93 , 0xa9 , 0xac , 0xa3 , 0xdb , 0x71 ,
+      0xcc , 0x30 , 0xc5 , 0x81 , 0x79 , 0xec , 0x3e , 0x87 ,
+      0xc1 , 0x4c , 0x01 , 0xd5 , 0xc1 , 0xf3 , 0x43 , 0x4f ,
+      0x1d , 0x87
+    },
+    82
+  },
+  {
+    {},
+    0,
+    {
+      0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b ,
+      0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b ,
+      0x0b , 0x0b , 0x0b , 0x0b , 0x0b , 0x0b
+    },
+    22,
+    {},
+    0,
+    {
+      0x19 , 0xef , 0x24 , 0xa3 , 0x2c , 0x71 , 0x7b , 0x16 ,
+      0x7f , 0x33 , 0xa9 , 0x1d , 0x6f , 0x64 , 0x8b , 0xdf ,
+      0x96 , 0x59 , 0x67 , 0x76 , 0xaf , 0xdb , 0x63 , 0x77 ,
+      0xac , 0x43 , 0x4c , 0x1c , 0x29 , 0x3c , 0xcb , 0x04
+    },
+    {
+      0x8d , 0xa4 , 0xe7 , 0x75 , 0xa5 , 0x63 , 0xc1 , 0x8f ,
+      0x71 , 0x5f , 0x80 , 0x2a , 0x06 , 0x3c , 0x5a , 0x31 ,
+      0xb8 , 0xa1 , 0x1f , 0x5c , 0x5e , 0xe1 , 0x87 , 0x9e ,
+      0xc3 , 0x45 , 0x4e , 0x5f , 0x3c , 0x73 , 0x8d , 0x2d ,
+      0x9d , 0x20 , 0x13 , 0x95 , 0xfa , 0xa4 , 0xb6 , 0x1a ,
+      0x96 , 0xc8
+    },
+    42
+  }
+};
+
+/*---------------------------------------------------------------------------*/
+UNIT_TEST_REGISTER(sha_256_hash_stepwise, "SHA-256 stepwise hashing");
+UNIT_TEST(sha_256_hash_stepwise)
+{
+  uint8_t sha256[SHA_256_DIGEST_LENGTH];
+  int i, j;
+  size_t len;
+  uint8_t buf[256];
+
+  UNIT_TEST_BEGIN();
+
+  for(i = 0; i < sizeof(hashes) / sizeof(hashes[0]); i++) {
+    SHA_256.init();
+    for(j = 0; j < sizeof(hashes[i].data) / sizeof(hashes[i].data[0]); j++) {
+      if(!hashes[i].data[j]) {
+        continue;
+      }
+      len = strlen(hashes[i].data[j]);
+      memcpy(buf, hashes[i].data[j], len);
+      SHA_256.update(buf, len);
+    }
+
+    SHA_256.finalize(sha256);
+    UNIT_TEST_ASSERT(!memcmp(sha256, hashes[i].hash, sizeof(sha256)));
+  }
+
+  UNIT_TEST_END();
+}
+/*---------------------------------------------------------------------------*/
+UNIT_TEST_REGISTER(sha_256_hash_with_checkpoint, "SHA-256 with checkpoint");
+UNIT_TEST(sha_256_hash_with_checkpoint)
+{
+  uint8_t sha256[SHA_256_DIGEST_LENGTH];
+  int i, j;
```

Comment:
I think both these can be declared in the head of the for-loop they belong to, and they should probably be `size_t` to avoid signed/unsigned comparisons.

Same comment for the loop counters in the other tests.